lib/logflare/logs/ingest_transformers.ex

```diff
@@ -73,4 +80,7 @@ defmodule Logflare.Logs.IngestTransformers do
         key
     end)
   end
+
+  # passthrough for non-matching data types
+  defp do_key_transform(log_params, _rule), do: log_params
```

Comment:
Yeah for now it's okay for this to blow up, or we should handle the error and respond appropriately to the client.  If this goes through it's just going to blow up down the line.

Reply:
But if we're moving towards allowing users to send in any value that they want for each top level key, it would not make sense to force the `metadata` key to be special anymore.
The only critical fields are `timestamp` and `event_message`, after all. All other fields can be treated any normal metadata field.

Reply:
Yeah I know I said I was okay with rewriting the BigQuery pipeline in place the last time we spoke but I think I was in an abnormally good mood or something.

I still very much think it should be rewritten as a "backend". 100% bad things will happen with an in place rewrite, even if done in small chunks.

I know the requirements and edge cases. Maybe it'll be approachable if we sit down and enumerate a v1 in more detail.

Reply:
hmm ok, i understand where you come from. lets focus on the v2 backend then. Just a bit worried that we'll miss edge cases from the original v1 code and spend extra time hunting them down.